### DISCUSSION TITLE
This could be considered as a DoS attack too?

### DISCUSSION THREAD

**Discussion by pymachado** [Cyfrin/security-and-auditing-full-course-s23#106] [Q&A] [answered]
In **Lesson 34: Mishandling of ETH: Minimized** in the PuppyRaffle security review, we checked a line of code like the one below:

```javascript
require(address(this).balance == uint256(totalFees), "PuppyRaffle: There are currently players active!");
```
When the vulnerability is exploited by a Smart Contract attacker who uses `selfdestruct()` to force the victim contract to receive ether, it renders the above instruction unusable because now `address(this).balance > uint256(totalFees)`. This results in the contract becoming unusable because you can't call the `PuppRaffe::withdraw` function again. Therefore, this can be considered a DoS attack, as users have been denied the service of withdrawal?

**Comment by PatrickAlphaC** [comment, marked as answer]
Sure could be :) 

Often, one type of attack could fall in many categorizations of attacks.

Way to think critically on this!

**Reply by pymachado** [reply]
Okay @PatrickAlphaC, thank you for your feedback
